llvm/include/llvm/Transforms/IPO/Attributor.h

```diff
@@ -3706,7 +3706,7 @@ struct AANoRecurse
   static AANoRecurse &createForPosition(const IRPosition &IRP, Attributor &A);
 
   /// See AbstractAttribute::getName()
-  const std::string getName() const override { return "AANoRecurse"; }
+  StringRef getName() const override { return "AANoRecurse"; }
```

Comment:
This probably should be something set in the AA constructor rather than an override